azurerm/resource_arm_backup_protected_file_share.go

```diff
@@ -0,0 +1,292 @@
+	if err != nil {
+		return fmt.Errorf("[ERROR] Unable to parse source_storage_account_id '%s': %+v", storageAccountID, err)
+	}
+	accountName, hasName := parsedStorageAccountID.Path["storageAccounts"]
+	if !hasName {
+		return fmt.Errorf("[ERROR] parsed source_storage_account_id '%s' doesn't contain 'storageAccounts'", storageAccountID)
+	}
+
+	protectedItemName := fmt.Sprintf("AzureFileShare;%s", fileShareName)
+	containerName := fmt.Sprintf("StorageContainer;storage;%s;%s", parsedStorageAccountID.ResourceGroup, accountName)
+
+	log.Printf("[DEBUG] Creating/updating Recovery Service Protected File Share %q (Container Name %q)", protectedItemName, containerName)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err2 := client.Get(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, "")
+		if err2 != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Recovery Service Protected File Share %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err2)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_backup_protected_file_share", *existing.ID)
+		}
+	}
+
+	item := backup.ProtectedItemResource{
+		Properties: &backup.AzureFileshareProtectedItem{
+			PolicyID:          &policyID,
+			ProtectedItemType: backup.ProtectedItemTypeAzureFileShareProtectedItem,
+			WorkloadType:      backup.DataSourceTypeAzureFileShare,
+			SourceResourceID:  utils.String(storageAccountID),
+			FriendlyName:      utils.String(fileShareName),
+		},
+	}
+
+	resp, err := client.CreateOrUpdate(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, item)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Recovery Service Protected File Share %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
+	}
+
+	locationURL, err := resp.Response.Location()
+	if err != nil || locationURL == nil {
+		return fmt.Errorf("Error creating/updating Azure File Share backup item %q (Vault %q): Location header missing or empty", containerName, vaultName)
+	}
+
+	opResourceID := azure.HandleAzureSdkForGoBug2824(locationURL.Path)
+
+	parsedLocation, err := azure.ParseAzureResourceID(opResourceID)
+	if err != nil {
+		return err
+	}
+	operationID := parsedLocation.Path["operationResults"]
+
+	if _, err := resourceArmBackupProtectedFileShareWaitForOperation(ctx, opClient, vaultName, resourceGroup, operationID, d); err != nil {
+		return err
+	}
+
+	resp, err = client.Get(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, "")
+
+	if err != nil {
+		return fmt.Errorf("Error creating/udpating Azure File Share backup item %q (Vault %q): %+v", protectedItemName, vaultName, err)
+	}
+
+	id := strings.Replace(*resp.ID, "Subscriptions", "subscriptions", 1) // This code is a workaround for this bug https://github.com/Azure/azure-sdk-for-go/issues/2824
```

Comment:
I believe this isn't required anymore (the service team claims its been fixed https://github.com/Azure/azure-sdk-for-go/issues/2824#issuecomment-563188743)